cpp/arcticdb/version/version_utils.hpp

```diff
@@ -160,6 +160,33 @@ inline void check_is_version(const AtomKey &key) {
     util::check(key.type() == KeyType::VERSION, "Expected version key type but got {}", key);
 }
 
+inline void verify_stream_id(const StreamId &stream_id) {
+    if (ConfigsMap::instance()->get_int("VersionStore.NoStrictSymbolCheck"))
+    {
+        ARCTICDB_DEBUG(log::version(), "Key with stream id {} will not be strictly checked because VersionStore.NoStrictSymbolCheck variable is set to 1.", stream_id);
+        return;
+    }
+
+    util::variant_match(
+        stream_id,
+        [] (const NumericId& num_stream_id) {
+            (void)num_stream_id; // Suppresses -Wunused-parameter
```

Comment:
Minor: if you remove the parameter name from the lambda (but leave the type), the compiler won't warn about it.

Reply:
The way we generally handle this is to use the macro ARCTICDB_UNUSED, which gives you the appropriate attribute for the platform